base/error.jl

```diff
@@ -52,28 +52,31 @@ end
 
 
 """
-    retry(f, [condition]; n=3; max_delay=10) -> Function
```

Comment:
Your points do make sense w.r.t. workloads which have some I/O component in them, think fetching S3 files, database reads, etc. On workloads which are all compute bound on large clusters, errors due to node deaths, or workers being unreachable, an immediate single retry makes sense.

I am not really sure what to go with here and hence chose the minimum option, leaving it upto programmers to specify values relevant to their scenarios.